CMakeLists.txt

```diff
@@ -26,6 +26,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
+if(MSVC)
+  set(COMPILE_WARNS /W4 /WX)
+else()
+  set(COMPILE_WARNS -g -Wall -Wextra -pedantic)
```

Comment:
why -g here? -g generates debug symbols. you control that on cmake via the CMAKE_BUILD_TYPE
the rest is ok.

Reply:
Edited, it is just warns now.